File: Modules/Common/CSM/SIE.Web.CSM/Suppliers/Commands/SelectUserCommand.cs
using SIE.CSM.Suppliers;
using SIE.Domain;
using SIE.Web.Command;
using System;
using System.Collections.Generic;

namespace SIE.Web.CSM.Suppliers.Commands
{
    /// <summary>
    /// 供应商选择物料关系
    /// </summary>
    public class SelectUserCommand : ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            if (args != null)
            {
                var meta = ClientEntities.Find(args.Type);//获取元数据
                var savedData = RF.Find(meta.EntityType).NewList();//新的数据集
                var supplierUserList = args.Data.ToJsonObject<List<SupplierUser>>();
                Check.NotNullOrEmpty(supplierUserList, nameof(supplierUserList));//检查空
                if (null == supplierUserList || supplierUserList.Count == 0)
                {
                    throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(supplierUserList)));
                }
                foreach (var item in supplierUserList)
                {
                    var supplierUser = new SupplierUser();
                    supplierUser.UserId = item.UserId;//用户
                    supplierUser.SupplierId = item.SupplierId;//供应商
                    savedData.Add(supplierUser);
                }
                RF.Save(savedData);//保存数据
            }
            return true;
        }
    }
}

File: Modules/Common/CSM/SIE.Web.CSM/Suppliers/Commands/SelectUserCommand.js
SIE.defineCommand('SIE.Web.CSM.Suppliers.Commands.SelectUserCommand', {
    extend: 'SIE.cmd.LookupCommandBase',
    userConfig: {
        dataParams: { specKeyPrototyName: 'UserId', targetClassName: 'SIE.Rbac.Users.User' },
    },
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    _flag: true,
    save: function (win) {
        var me = this;
        var indata = {};
        var selections = me._targetSelectItems;
        if (me._flag) {
            if (selections && selections.items.length > 0) {
                me._flag = false;
                var operationDatas = [];
                SIE.each(selections.items, function (item) {
                    var userId = item.getId();
                    if (me._sourceViewSelectItems.indexOf(userId) === -1) {
                        var supplierUser = { SupplierId: me._sourceId, UserId: userId };
                        operationDatas.push(supplierUser);
                    }
                });
                indata = operationDatas;
                me._targetView.execute({
                    data: indata,
                    success: function (res) {
                        me._flag = true;
                        win.close();  //关闭模态窗口
                        me._ownerView.loadChildData(true); //重载视图数据                    
                    },
                    error: function () {
                        me._flag = true;
                    }
                }, me._ownerView);
            }
            else {
                SIE.Msg.showWarning('没有可提交的数据'.t());
            }
        }
    }
    // end 
});